--- archiso.1	2021-06-23 01:17:18.815644100 +0800
+++ archiso	2021-06-23 01:18:41.151815377 +0800
@@ -67,7 +67,15 @@
 
     # shellcheck disable=SC2154
     # defined via initcpio's parse_cmdline()
-    if [ "${copytoram}" = "y" ]; then
+    if [ -n "${ip}" ] && [ -n "${archiso_http_srv}" ]; then
+        img=/run/archiso/copytoram/"${img_fullname}"
+        msg -n ":: Moving squashfs image in RAM..."
+        umount /run/archiso/copytoram
+        mount --move /run/archiso/httpspace /run/archiso/copytoram
+        mv /run/archiso/copytoram/"${archisobasedir}/${arch}/${img_fullname}" "$img"
+        rm -rf /run/archiso/copytoram/"${archisobasedir}"
+        msg "done."
+    elif [ "${copytoram}" = "y" ]; then
         msg -n ":: Copying squashfs image to RAM..."
 
         # in case we have pv use it to display copy progress feedback otherwise
